drizzle/utils.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+
+
+def calc_pixmap(wcs_from, wcs_to, estimate_pixel_scale_ratio=False,
+                refpix_from=None, refpix_to=None):
+    """
+    Calculate a mapping between the pixels of two images. Pixel scale ratio,
+    when requested, is computed near the centers of the bounding box
+    (a property of the WCS object) or near ``refpix_*`` coordinates
+    if supplied.
+
+    Parameters
+    ----------
+
+    wcs_from : wcs
+        A WCS object representing the coordinate system you are
+        converting from. This object *must* have ``pixel_shape`` property
+        defined.
+
+    wcs_to : wcs
+        A WCS object representing the coordinate system you are
+        converting to.
+
+    estimate_pixel_scale_ratio : bool, optional
+        Estimate the ratio of "to" to "from" WCS pixel scales.
+
+        .. note::
+            Pixel scale is estimated as the square root of pixel's area
+            (i.e., pixels are assumed to have a square shape) at the reference
+            pixel position which is taken as the center of the bounding box
+            if ``wcs_*`` has a bounding box defined, or as the center of the box
+            defined by the ``pixel_shape`` attribute of the input WCS if
+            ``pixel_shape`` is defined (not `None`), or at pixel coordinates
+            ``(0, 0)``.
+
+    refpix_from : numpy.ndarray, tuple, list
+        Image coordinates of the reference pixel near which pixel scale should
+        be computed in the "from" image. In FITS WCS this could be, for example,
+        the value of CRPIX of the ``wcs_from`` WCS.
+
+    refpix_to : numpy.ndarray, tuple, list
+        Image coordinates of the reference pixel near which pixel scale should
+        be computed in the "to" image. In FITS WCS this could be, for example,
+        the value of CRPIX of the ``wcs_to`` WCS.
+
+    Returns
+    -------
+
+    pixmap : numpy.ndarray
+        A three dimensional array representing the transformation between
+        the two. The last dimension is of length two and contains the x and
+        y coordinates of a pixel center, repectively. The other two coordinates
+        correspond to the two coordinates of the image the first WCS is from.
+
+    pixel_scale_ratio : float
+        Estimate the ratio of "to" to "from" WCS pixel scales. This value is
+        returned only when ``estimate_pixel_scale_ratio`` is `True`.
+
+    """
+    # We add one to the pixel co-ordinates before the transformation and subtract
+    # it afterwards because wcs co-ordinates are one based, while pixel co-ordinates
+    # are zero based, The result is the final values in pixmap give a tranformation
+    # between the pixel co-ordinates in the first image to pixel co-ordinates in the
+    # co-ordinate system of the second.
```

Comment:
Should the above comment be deleted?

Reply:
Of course it will be deleted. At this stage this PR is about API only. However, since it looks like you and @perrygreenfield approve this API redesign, I will now work on cleaning up comments and docstrings.

Reply:
I forgot about this. will be fixed